tests/php/ORM/DatabaseTest.php

```diff
@@ -345,4 +372,45 @@ public function testRepeatedPartialIteration()
             $i++;
         }
     }
+
+    /**
+     * Test that stopping iteration part-way through produces predictable results even when we're using predicates
+     * on a subsequent iteration.
+     * This test is here to ensure consistency between implementations (e.g. mysql vs postgres, etc)
+     */
+    public function testRepeatedPartialIterationWithPredicates()
```

Comment:
This test is less important but just validates that predicated queries behave the same way as non-predicated queries for partial iterations.